mean-slopes/Lines_Slopes_and_Intercepts_-_Refresher.Rmd

```diff
@@ -0,0 +1,581 @@
+```{python}
+# do not worry about any of the code in this notebook, it is just here to generate the graphs
+y1 = 5*x
+plt.figure(figsize = (8,8))
+plt.plot(x,y, color = 'red', label = 'y = 1 * x')
+plt.plot(x,y1, color = 'blue', label = 'y = 5 * x')
+plt.plot(x_axis, np.repeat(0, len(x_axis)), color = 'black')
+plt.plot(np.repeat(0, len(y_axis)), y_axis, color = 'black')
+plt.xlabel('X Axis')
+plt.ylabel('Y Axis')
+plt.ylim(min(y_axis), max(y_axis))
+plt.xticks(np.arange(np.min(x_axis), np.max(x_axis)+1, 1))
+plt.yticks(np.arange(np.min(y_axis), np.max(y_axis)+1, 1))
+plt.plot(0,0, 'o', color = 'black', markersize = 8, label = 'The Origin (x = 0, y = 0)')
+plt.legend()
+plt.show()
+```
+
+The number we have changed -  from '1' to '5' - is the <b> slope </b>. Graphically, we can see that changing the slope changes the steepness of the line : the line for ```y = 5 * x``` is much steeper than the line for ```y = 1 * x```.
+
+What the slope tells us is this: for the x-coordinate of any point on the line, how much bigger/smaller is the y coordinate?
+
+In the case of ```y = 1  * x```: 
+
+where x = 1, y equals 1 times the value of x ... so y = 1,
+
+where x = 5, y equals 1 times the value of x ... so y = 5 etc.
+
+In the case of ```y = 5 * x```, y equals 5 times the value of x. So: 
+
+where x = 1, y equals 5 times the value of x ... so y = 5,
+
+where x = 5, y equals 5 times the value of x... so y = 25 etc.
+
+The table below shows this for a set of points on the line ```y = 5 * x```:
+
+```{python}
+df = pd.DataFrame({'x coordinate:': x, 'y coordinate:': y1})
+df.index = np.repeat('', len(df))
+display(df)
+```
+
+Another way of thinking of this is: if we take any two points on the line, and we calculate how much the x-value has changed between those points, then the slope tells us how much the y value has also changed.
+
+So for the line ```y = 1 * x```, if we follow the line from x = 0 to x = 5, the slope tells us how much the y values have changed over this distance. 
+
+The graph below shows that, for the line ```y = 1 * x```, as x changes from 0 to 5, y also changes from 0 to 5.
+
+(This graph shows slightly more of the x and y axes than the previous graphs, as we will need this for the line we plot after this one).
+
+```{python}
+# do not worry about any of the code in this notebook, it is just here to generate the graphs
+plt.figure(figsize = (8,8))
+y_axis_ext = np.arange(-30, 31)
+x_axis_ext = np.arange(-30, 31)
+plt.plot(x_axis_ext , np.repeat(0, len(x_axis_ext)), color = 'black')
+plt.plot(np.repeat(0, len(y_axis_ext)), y_axis_ext, color = 'black')
+plt.plot(x,y, color = 'red', label = 'y = 1 * x')
+plt.plot([5, 0, 5], [5,5, 0], 'x', color = 'black')
+zer_5 = np.arange(0,6)
+plt.plot(np.repeat(5, len(zer_5)), zer_5, '--', color = 'darkgreen', label = 'Change in y')
+plt.plot(zer_5, np.repeat(5, len(zer_5)),'--', color = 'cyan', label = 'Change in x')
+plt.xlabel('X Axis')
+plt.ylabel('Y Axis')
+plt.ylim(min(y_axis_ext), max(y_axis_ext))
```

Comment:
Maybe a function to do this axis extent stuff - if you are re-using the same code for many cells?